packages/collections/persistent/test.pony

```diff
@@ -0,0 +1,391 @@
+    //1100 00011 11101 01001 10111 or 12711223
+    let f0 = _BitOps.arrayIdxFor(12711223, 0)
+    let f1 = _BitOps.arrayIdxFor(12711223, 5)
+    let f2 = _BitOps.arrayIdxFor(12711223, 10)
+    let f3 = _BitOps.arrayIdxFor(12711223, 25)
+    h.assert_eq[USize](f0, 0)
+    h.assert_eq[USize](f1, 4)
+    h.assert_eq[USize](f2, 6)
+    h.assert_eq[USize](f3, 14)
+
+    true
+
+class iso _TestHAMTMap is UnitTest
+  fun name(): String => "persistent-data/Map"
+
+  fun apply(h: TestHelper) =>
+    let m1: Map[String,U32] = Maps.empty[String,U32]()
+    h.assert_error(lambda()(m1)? => m1("a") end)
+    let s1 = m1.size()
+    h.assert_eq[U64](s1, 0)
+
+    try
+      let m2 = m1.update("a", 5)
+      let m3 = m2.update("b", 10)
+      let m4 = m3.update("a", 4)
+      let m5 = m4.update("c", 0)
+      h.assert_eq[U32](m2("a"), 5)
+      h.assert_eq[U32](m3("b"), 10)
+      h.assert_eq[U32](m4("a"), 4)
+      h.assert_eq[U32](m5("c"), 0)
+    else
+      h.complete(false)
+    end
+
+    try
+      let m6 = Maps.from[String,U32]([("a", 2), ("b", 3), ("d", 4), ("e", 5)])
+      let m7 = m6.update("a", 10)
+      h.assert_eq[U32](m6("a"), 2)
+      h.assert_eq[U32](m6("b"), 3)
+      h.assert_eq[U32](m6("d"), 4)
+      h.assert_eq[U32](m6("e"), 5)
+      h.assert_eq[U32](m7("a"), 10)
+      h.assert_eq[U32](m7("b"), 3)
+      h.assert_eq[U32](m7("a"), 10)
+      let m8 = m7.remove("a")
+      h.assert_error(lambda()(m8 = m8)? => m8("a") end)
+      h.assert_eq[U32](m8("b"), 3)
+      h.assert_eq[U32](m8("d"), 4)
+      h.assert_eq[U32](m8("e"), 5)
+      let m9 = m7.remove("e")
+      h.assert_error(lambda()(m9 = m9)? => m9("e") end)
+      h.assert_eq[U32](m9("b"), 3)
+      h.assert_eq[U32](m9("d"), 4)
+      let m10 = m9.remove("b").remove("d")
+      h.assert_error(lambda()(m10 = m10)? => m10("b") end)
+      h.assert_error(lambda()(m10 = m10)? => m10("d") end)
+    else
+      h.complete(false)
+    end
+
+    true
+
+class iso _TestMapVsMap is UnitTest
+  fun name(): String => "persistent-data/Map vs. collections Map"
+
+  fun apply(h: TestHelper) ? =>
+    var pMap: Map[String,U64] = Maps.empty[String,U64]()
+    let mMap: mut.Map[String,U64] = mut.Map[String,U64]()
+    let kvs = Array[(String,U64)]()
+    let dice = Dice(MT)
+    var count: USize = 0
+    let iterations: USize = 100000
+    let keys: U64 = 10000
+
+    while(count < iterations) do
+      let k0 = dice(1,keys).string()
+      let k: String val = consume k0
+      let v = dice(1,100000)
+      kvs.push((k, v))
+      count = count + 1
+    end
+
+    count = 0
+    while(count < iterations) do
+      let k = kvs(count)._1
+      let v = kvs(count)._2
+      pMap = pMap.update(k, v)
+      mMap.update(k, v)
+      count = count + 1
+    end
+    count = 0
+    while(count < iterations) do
+      let pmv = pMap(kvs(count)._1)
+      let mmv = mMap(kvs(count)._1)
+      h.assert_eq[Bool](H.equalMapU64Values(pmv, mmv), true)
+      count = count + 1
+    end
+
+    true
+
+primitive H
```

Comment:
Note that this would be a public type in the package, which is probably not what you want since it's here in `test.pony`.  You probably want to prefix this with an underscore.